### PR TITLE
feat: Disable language analysis for a repository DOCS-686

### PR DESCRIPTION
**Pull request by claudiacarpinteiro**
Updated product documentation to reflect the option to disable the analysis of a specific language for a repository.

### :eyes: Live preview

- https://docs-686-disable-language--docs-codacy.netlify.app/repositories-configure/languages/
- https://docs-686-disable-language--docs-codacy.netlify.app/organizations/roles-and-permissions-for-organizations/
- https://docs-686-disable-language--docs-codacy.netlify.app/repositories-configure/codacy-configuration-file/

### :construction: To do
-   [x] Update content considering the new behavior
-   [x] Update screenshots
-   [x] Request validation
-   [x] Apply feedback
-   [x] Wait for feature release
-   [x] Remove **don't merge** label before merging

### CODE REVIEW

**Review by nicklem — COMMENTED** [review]
Looking good! Left a couple tweaks.

**Review by pedrocodacy — APPROVED** [review]
On the side of engineering this looks great :)

Approving now, but I see that there are some corrections on the text suggested, if this ends up having changes feel free to ping me for a re-approve 🙏

**Review by lorandszakacs — APPROVED** [review]

File: docs/repositories-configure/codacy-configuration-file.md [hunk 1]
```diff
@@ -249,3 +252,37 @@ engines:
     ignoreAnnotations: false
     ignoreUsings: false
 ```
+
+## Configuring languages using a Codacy configuration file {: id="languages"}
+
+You can use a Codacy configuration file to manage the languages that Codacy analyzes in your repository.
+
+!!! note
+    Codacy applies both language settings defined in the Codacy configuration file and any other language setting defined [on the Codacy UI](languages.md).
```
[the remark this answers is not in the document]
Comment (claudiacarpinteiro) [reply docs/repositories-configure/codacy-configuration-file.md:261]: > Note: it's also unclear "how" these settings are applied. Which ones take precedence?

@nicklem AFAIK, there's no precedence, they are cumulative:
- If we have different custom extensions defined in the configuration file and on the Codacy UI, all of them will be applied. 
- All languages [are enabled by default](https://github.com/codacy/docs/pull/2112/files#diff-6e1b2afc217b0798259f3b0065e205a68fd5c13f20e03dae4014a2c39509173aR279). Thus, Codacy will not analyze the languages disabled in the configuration file, nor the languages disabled on the Codacy UI. Makes sense? 🤔 

@pedrocodacy Can you confirm if there is no precedence for these configurations?

[the remark this answers is not in the document]
Comment (claudiacarpinteiro) [reply docs/repositories-configure/codacy-configuration-file.md:261]: @nicklem Did you remove the link for a particular reason? Let me know if this works:

```suggestion
    Codacy applies the language settings from the Codacy configuration file as well as any settings defined [in the Codacy UI](languages.md).
```

File: docs/repositories-configure/languages.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Codacy automatically detects and analyzes languages in your repository based on a list of supported file extensions. On the repository **Settings**, tab **Languages**, you can see the list of languages that Codacy detected and will include in the next analysis.
+
+By default, the analysis is enabled for all languages that Codacy detects in you repository. You can adapt this configuration to your needs as follows:
+
+-   [Configure Codacy to include unrecognized extensions](#configuring-file-extensions)
+-   [Disable the analysis of specific languages](#disable-language)
+
+!!! note
+    Codacy applies both **Languages** settings defined on the Codacy UI and any other language setting defined using a [Codacy configuration file](codacy-configuration-file.md#languages).
```
[the remark this answers is not in the document]
Comment (claudiacarpinteiro) [reply docs/repositories-configure/languages.md:17]: Adding a tiny tweak to your suggestion:

```suggestion
    Codacy applies the language settings defined in the Codacy UI as well as any settings defined using a [Codacy configuration file](codacy-configuration-file.md#languages).
```

File: docs/repositories-configure/codacy-configuration-file.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicklem) [reply docs/repositories-configure/codacy-configuration-file.md:261]: Thanks @claudiacarpinteiro, all clear!

> Did you remove the link for a particular reason?
I believe the autocorrect removed it 😅 let's keep it, yes

[the remark this answers is not in the document]
Comment (claudiacarpinteiro) [reply docs/repositories-configure/codacy-configuration-file.md:261]: Keeping your suggestion and avoid "both".

```suggestion
    Codacy applies the language settings from the Codacy configuration file as well as any settings defined [in the Codacy UI](languages.md).
```

**Pull request merged**